onnxruntime/contrib_ops/cpu/word_conv_embedding.cc

```diff
@@ -51,53 +52,65 @@ void WordConvEmbedding::ComputeConvMaxPoolWithActivation(
   int64_t conv_res_segment_size = unfolded_width * num_filters;
   int64_t memcpy_size = unfolded_kernal_size * sizeof(float);
 
-  auto input_unfolded_buffer_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * unfolded_segment_size);
+  auto unfolded_buffer_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * unfolded_segment_size);
   auto conv_result_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * conv_res_segment_size);
   auto conv_activation_result_p = IAllocator::MakeUniquePtr<float>(allocator, seq_len * conv_res_segment_size);
 
-  for (int64_t word_inx = 0; word_inx < seq_len; word_inx++) {
-    if (words_len_ptr[word_inx] <= 0) continue;
-
-    const float* current_word_input = input + word_inx * input_word_size;
-    float* current_word_unfolded_buffer_p = input_unfolded_buffer_p.get() + word_inx * unfolded_segment_size;
-    float* conv_buf_p = conv_result_p.get() + word_inx * conv_res_segment_size;
-    float* pactivationbuf = conv_activation_result_p.get() + word_inx * conv_res_segment_size;
-    float* pres = output + word_inx * num_filters;
-
-    // Unfolding from pin to pufbuf.
-    float* tmp_unfolded_buffer_ptr = current_word_unfolded_buffer_p;
-    for (int64_t unfolded_inx = 0; unfolded_inx < unfolded_width; unfolded_inx++) {
-      memcpy(tmp_unfolded_buffer_ptr, current_word_input, memcpy_size);
-      current_word_input += char_embedding_size;
-      tmp_unfolded_buffer_ptr += unfolded_kernal_size;
+  int64_t word_inx = 0;
+  while (word_inx < seq_len) {
+    if (words_len_ptr[word_inx] <= 0) {
+      word_inx++;
+      continue;
+    }
+
+    float* words_unfolded_buffer_p = unfolded_buffer_p.get();
+    int64_t words_unfolded_width = 0;
+    int64_t tmp_word_inx = word_inx;
+    float* conv_buf_p = conv_result_p.get();
+    float* pactivationbuf = conv_activation_result_p.get();
+
+    // unfolding buffer
+    while (words_len_ptr[tmp_word_inx] > 0 && tmp_word_inx < seq_len) {
```

Comment:
Maybe change the order of the two conditions in case words_len_ptr.size() == seq_len?